### ISSUE TITLE
Crash on Android: BadTokenException

### ISSUE THREAD

**Issue by felixkrautschuk** [TobiasHennig/nativescript-toast#33]
Hey,

I cannot give much detailed information as this issue occurs only sometimes, but maybe someone else makes the same experience. The plugin works fine on iOS, but sometimes I get the following exception on Android:

```
android.view.WindowManager$BadTokenException: Unable to add window -- token android.os.BinderProxy@9441757 is not valid; is your activity running?
	  at android.view.ViewRootImpl.setView(ViewRootImpl.java:679)
	  at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:342)
	  at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:93)
	  at android.widget.Toast$TN.handleShow(Toast.java:459)
	  at android.widget.Toast$TN$2.handleMessage(Toast.java:342)
	  at android.os.Handler.dispatchMessage(Handler.java:102)
	  at android.os.Looper.loop(Looper.java:154)
	  at android.app.ActivityThread.main(ActivityThread.java:6121)
	  at java.lang.reflect.Method.invoke(Native Method)
	  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
	  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
```

My configuration:
```
{
  "nativescript": {
    "id": "net.fanfiktion.mobile",
    "tns-android": {
      "version": "3.4.2"
    },
    "tns-ios": {
      "version": "3.4.1"
    }
  },
  "dependencies": {
    "nativescript-toast": "1.4.6",
    "tns-core-modules": "^3.4.1"
  },
  "devDependencies": {
    "babel-traverse": "^6.26.0",
    "babel-types": "^6.26.0",
    "babylon": "^6.18.0",
    "lazy": "^1.0.11",
    "tns-platform-declarations": "^3.4.1"
  }
}
```

I hope anybody has an idea how to handle this issue.

Best regards,
Felix

**Comment by gsmedley** [comment]
Did you ever resolve this?  I'm seeing the same crash, but it's not easy to reproduce. Seems to happen starting a toast, going to another page (I'm using Angular) then maybe going back quickly.
<pre>
11-26 12:09:32.857  2611  2611 W WindowManager: Attempted to remove non-existing token: android.os.Binder@fed8d48
11-26 12:09:32.879  2611  2626 D WindowManager: addWindow: android.view.IWindow$Stub$Proxy@6e147e1 displayId=0
11-26 12:09:32.879  2611  2626 W WindowManager: Attempted to add a toast window with unknown token android.os.Binder@fed8d48.  Aborting.
11-26 12:09:32.882 20878 20878 D AndroidRuntime: Shutting down VM
11-26 12:09:32.884 20878 20878 W System.err: android.view.WindowManager$BadTokenException: Unable to add window -- token android.os.BinderProxy@d52c5ec is not valid; is your activity running?
11-26 12:09:32.884 20878 20878 W System.err: 	at android.view.ViewRootImpl.setView(ViewRootImpl.java:920)
11-26 12:09:32.884 20878 20878 W System.err: 	at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:377)
11-26 12:09:32.885 20878 20878 W System.err: 	at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:105)
11-26 12:09:32.885 20878 20878 W System.err: 	at android.widget.Toast$TN.handleShow(Toast.java:747)
11-26 12:09:32.885 20878 20878 W System.err: 	at android.widget.Toast$TN$2.handleMessage(Toast.java:622)
11-26 12:09:32.885 20878 20878 W System.err: 	at android.os.Handler.dispatchMessage(Handler.java:102)
11-26 12:09:32.885 20878 20878 W System.err: 	at android.os.Looper.loop(Looper.java:154)
11-26 12:09:32.885 20878 20878 W System.err: 	at android.app.ActivityThread.main(ActivityThread.java:6823)
11-26 12:09:32.885 20878 20878 W System.err: 	at java.lang.reflect.Method.invoke(Native Method)
11-26 12:09:32.885 20878 20878 W System.err: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1557)
11-26 12:09:32.885 20878 20878 W System.err: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1445)
11-26 12:09:32.893 20878 20878 E AndroidRuntime: FATAL EXCEPTION: main
11-26 12:09:32.893 20878 20878 E AndroidRuntime: Process: com.kanayo.mymobilenative2, PID: 20878
11-26 12:09:32.893 20878 20878 E AndroidRuntime: android.view.WindowManager$BadTokenException: Unable to add window -- token android.os.BinderProxy@d52c5ec is not valid; is your activity running?
11-26 12:09:32.893 20878 20878 E AndroidRuntime: 	at android.view.ViewRootImpl.setView(ViewRootImpl.java:920)
11-26 12:09:32.893 20878 20878 E AndroidRuntime: 	at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:377)
11-26 12:09:32.893 20878 20878 E AndroidRuntime: 	at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:105)
11-26 12:09:32.893 20878 20878 E AndroidRuntime: 	at android.widget.Toast$TN.handleShow(Toast.java:747)
11-26 12:09:32.893 20878 20878 E AndroidRuntime: 	at android.widget.Toast$TN$2.handleMessage(Toast.java:622)
11-26 12:09:32.893 20878 20878 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:102)
11-26 12:09:32.893 20878 20878 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:154)
11-26 12:09:32.893 20878 20878 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6823)
11-26 12:09:32.893 20878 20878 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
11-26 12:09:32.893 20878 20878 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1557)
11-26 12:09:32.893 20878 20878 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1445)
11-26 12:09:32.896  2611  2769 D Debug   : !@DumpState : SHIP
11-26 12:09:32.896  2611  2769 D Debug   : !@DumpState : debug level:0x4f4c
11-26 12:09:32.896  2611  2769 D Debug   : !@Dumpstate : Finally, system will skip dumpstate
11-26 12:09:32.897  2611  2769 W ActivityManager:   Force finishing activity com.kanayo.mymobilenative2/com.tns.NativeScriptActivity
</pre>

**Comment by hypery2k** [comment]
Should be fixed in [Version 2.0.0](https://github.com/TobiasHennig/nativescript-toast/releases/tag/v2.0.0).

If not reopen this issue

**Comment by karthick-in** [comment]
Hey @hypery2k 

I'm facing this issue in my nativescript angular application

Here's the Toast version
`"nativescript-toast": "^2.0.0",`

Package.json
```
"tns-ios": {
      "version": "6.5.0"
    },
    "tns-android": {
      "version": "6.5.3"
    }
```

Let me know if this issue has any workaround.
